app/controllers/posts_controller.rb

```diff
@@ -11,17 +13,28 @@ def show
   end
 
   def new
+    @user = User.find(params[:user_id])
     @post = Post.new
   end
 
   def create
-    @post = current_user.posts.new(post_params)
+    @post = @user.posts.new(post_params)
```

Comment:
- [ ] Nice work implementing the create action for the post :clap: However, according to the project requirement post should be created on behalf of the `current_user` (using the method that you created in your ApplicationController). Therefore I suggest that you change the `@user` to use the `current_user` method created in the ApplicationController